lib/pwa.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Progressive Web App
+ *
+ * @package gutenberg
+ */
+
+add_filter(
+	'admin_head',
+	function() {
+		$l10n = array(
+			'logo'      => file_get_contents( ABSPATH . 'wp-admin/images/wordpress-logo-white.svg' ),
+			'siteTitle' => get_bloginfo( 'name' ),
+			'adminUrl'  => admin_url(),
+		);
+		wp_enqueue_script( 'wp-admin-manifest' );
+		wp_localize_script( 'wp-admin-manifest', 'wpAdminManifestL10n', $l10n );
+	}
+);
+
+add_filter(
+	'load-index.php',
```

Comment:
The PWA plugin makes use of `admin-ajax.php` for serving the service worker which also works.

Reply:
Right, we could do that too :)